x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -197,15 +198,34 @@ const AssistantComponent: React.FC<Props> = ({
   }, [currentConversation?.title, setConversationTitle]);
 
   const refetchCurrentConversation = useCallback(
-    async ({ cId, cTitle }: { cId?: string; cTitle?: string } = {}) => {
+    async ({
+      cId,
+      cTitle,
+      isStreamRefetch = false,
+    }: { cId?: string; cTitle?: string; isStreamRefetch?: boolean } = {}) => {
       if (cId === '' || (cTitle && !conversations[cTitle])) {
         return;
       }
 
       const conversationId = cId ?? (cTitle && conversations[cTitle].id) ?? currentConversation?.id;
 
       if (conversationId) {
-        const updatedConversation = await getConversation(conversationId);
+        let updatedConversation = await getConversation(conversationId);
+        let retries = 0;
+        const maxRetries = 5;
+
+        // this retry is a workaround for the stream not YET being persisted to the stored conversation
+        while (
+          isStreamRefetch &&
+          updatedConversation &&
+          updatedConversation.messages[updatedConversation.messages.length - 1].role !==
+            'assistant' &&
+          retries < maxRetries
```

Comment:
Steph and I discussed this fix here, seems like the best approach at the moment given we need to re-fetch to retrieve replacements + trace data (could just update local conversation cache w/ stream result if we didn't need replacements). 

Would be nice if we had some sort of push notification service, but in lieu of that we've gotta poll for the auxiliary data post-stream. This poll up to 10s should handle any delays in serverless persistence as well.